core/management/commands/update_data.py

```diff
@@ -144,3 +149,23 @@ def handle(self, *args, **kwargs):
                 io.BytesIO(response_data), sheet_name=Model.__name__, workbook_kwargs={"read_only": False}
             )
             self._update_data(Model, table, update_data_function)
+
+        print("Updating DataTable...", end="", flush=True)
+        total_created, total_updated, total_skipped = 0, 0, 0
+        for table in Table.with_hidden.select_related("dataset"):
+            key = (table.dataset.slug, table.name)
+            try:
+                data_table = data_tables_map[key]
+                if data_table is None:
+                    raise ValueError
+                if data_table.table != table:  # Tables were truncated so previous DataTables get updated
+                    total_updated += 1
+                    data_table.table = table
+                    data_table.save()
+                else:  # Same table as before, so no need to update
+                    total_skipped += 1
+            except (KeyError, ValueError):  # create DataTable if new Table or if previous was None
+                data_table = DataTable.new_data_table(table, suffix_size=0)
+                data_table.activate()
+                total_created += 1
```

Comment:
Não sei se entendi bem a lógica, mas o seguinte código não resolveria?

```suggestion
            data_table = data_tables_map.get(key, None)
            if data_table is None:  # create DataTable if new Table or if previous was None
                data_table = DataTable.new_data_table(table, suffix_size=0)
                data_table.activate()
                total_created += 1
            elif data_table.table != table:  # Tables were truncated so previous DataTables get updated
                total_updated += 1
                data_table.table = table
                data_table.save()
            else:  # Same table as before, so no need to update
                total_skipped += 1
```

Reply:
Sim, total! Usando o `.get` fica mais simples mesmo